### PR TITLE
Client retry

### PR DESCRIPTION
**Pull request by alexcb**

### CODE REVIEW

File: secretsclient/client.go [hunk 1]
```diff
@@ -102,6 +114,36 @@ func (c *client) doCall(method, url string, opts ...requestOpt) (int, string, er
...
+			}
+		}
+		if duration > maxSleepBeforeRetry {
+			duration = maxSleepBeforeRetry
+		}
+		time.Sleep(duration)
+		duration *= 2
+	}
+	return status, body, err
+}
```
Comment (vladaionescu) [inline secretsclient/client.go:144]: Wondering if we should use some library for this. Although this is pretty small and simple too.

File: secretsclient/client.go [hunk 2]
```diff
@@ -143,16 +185,18 @@ type client struct {
 	sshKey                string
 	lastUsedPublicKeyPath string
 	sshAgent              agent.ExtendedAgent
+	warnFunc              func(string, ...interface{})
```
Comment (vladaionescu) [inline secretsclient/client.go:188]: Maybe it's simpler to just pass the entire `ConsoleLogger` object? But not a strong opinion.

**Review by vladaionescu — APPROVED** [review]

File: secretsclient/client.go (hunk 2 above)
Comment (alexcb) [reply secretsclient/client.go:188]: I went with this as it gives us an option to move this client into an outside library down the road.

Comment (vladaionescu) [reply secretsclient/client.go:188]: Good idea

File: secretsclient/client.go (hunk 1 above)
Comment (alexcb) [reply secretsclient/client.go:144]: I was half tempted to use one, but figured less (dependencies) is more.

**Pull request merged**